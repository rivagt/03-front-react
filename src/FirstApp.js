import PropTypes from 'prop-types';

//functional components
const FirstApp = ( {saludo, subtitulo} ) => {
    return (
        //div es para darle css
        <>
            <h1> {saludo} </h1>
        {/* no se pueden usar booleans, concatena arrays pero los separa individualmente, no imprime objetos */}
            {/* <pre>{ JSON.stringify(saludo, null, 3) }</pre> */}
            <p> {subtitulo} </p>
        </>
    );
}

FirstApp.propTypes = {
    saludo: PropTypes.string.isRequired
}
FirstApp.defaultProps = {
    subtitulo: 'No soy un subtitulo'
}

export default FirstApp; 