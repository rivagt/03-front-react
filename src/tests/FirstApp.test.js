// import { render } from '@testing-library/react';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';
import FirstApp from '../FirstApp';


describe('Test in component FirstApp', () => {
    test('mostrar <FirstApp /> correctly', () => {
        const saludo = 'Hi im Rivita';
        const wrapper = shallow(<FirstApp saludo={saludo}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('show me subtitle from props', () => {
        const saludo = 'Hi Im Riva';
        const subtitle = 'Subtitle from FirstApp.test';
        const wrapper = shallow(<FirstApp saludo={ saludo } subtitulo={ subtitle }/>);
        const textp = wrapper.find('p').text().trim();
        // console.log(textp);
        expect(textp).toBe(subtitle);
    });
    
    
    
    //Test with import '@testing-library/jest-dom/extend-expect';
    // test('debe retornar soy goku', () => {
    //     const saludo = 'Hi Im Riita';
    //     const {getByText} = render(<FirstApp saludo={saludo}/>);

    //     expect(getByText(saludo)).toBeInTheDocument();
    // });
    
})
