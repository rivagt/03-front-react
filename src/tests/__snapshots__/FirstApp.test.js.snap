// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in component FirstApp mostrar <FirstApp /> correctly 1`] = `
<Fragment>
  <h1>
     
    Hi im Rivita
     
  </h1>
  <p>
     
    No soy un subtitulo
     
  </p>
</Fragment>
`;
